R/read_yld.R

```diff
@@ -4,7 +4,64 @@
 #' This file contains yield information.
 #'
 #' @param file A path to the file.
+#' @export
 #'
 read_yld <- function(file) {
-  warning("This function is not yet implemented.")
+  d <- readLines(file)
+
+  yields= c()
+  crops = c()
+
+  blank_line = 0
+  line = 7
+
+
+  while (blank_line < 1) {
+    tmp = strsplit(d[line], split = ' ')[[1]]
+
+    if (length(tmp) >= 4) {
+      crop_id = as.numeric(tmp[6])
+      crop_type = tmp[8]
+
+      crops = rbind(crops, c(crop_id, crop_type))
+    } else {
+      blank_line = blank_line + 1
+    }
+    line = line + 1
+  }
+
+  # from the previous loop
+  line = line + 2
+
+  while (blank_line < 2) {
+    tmp = strsplit(d[line], split = ' ')[[1]]
+
+    # assure that NOT a blank line
+    if (length(tmp) >= 12) {
+      blank_line = 0
+
+      crop_type = as.numeric(tmp[6])
+      date = as.numeric(tmp[9])
+      OFE = as.numeric(tmp[13])
+      yield = as.numeric(tmp[18])
+      unit = tmp[19]
+      year = as.numeric(tmp[21])
+
+      row = c(crop_type, date, OFE, yield, unit, year)
```

Comment:
All variables are converted to character here.

Reply:
So I believe I shouldn't merge this until you fix the type conversion. Correct? @kgulzina

Reply:
yes, @jarad

Reply:
@jarad, I fixed the type conversion. Can merge now.